offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/QuotationDetails.groovy

```diff
@@ -155,33 +208,38 @@ class QuotationDetails {
      */
     private void generateDiscountItemContent(String description, String elementId, OfferItem item){
         itemNumber++
-        if (isOverflowingPage()) {
+        if (isTableOverflowByItem(item)) {
             generateHTMLTableOnNextPage(elementId)
             //repeat table header for next page
             htmlContent.getElementById(elementId).append(ItemPrintout.tableHeader())
             resetPageItemsCount()
+            consumedPageSpace += TableElementSpacing.TABLE_HEADER.getTableElementSpacing()
         }
-        htmlContent.getElementById(elementId).append(ItemPrintout.discountItemInHTML(itemNumber, description, item))
-        pageItemsCount += determineItemSpace("Discount", description, item.quantityDiscount)
+
+        String quantity = item.quantity as String
+        String unit = item.unit as String
+        String unitDiscount = Currency.getFormatterWithoutSymbol().format(item.getDiscountPerUnit())
+        String discountQuantity = Currency.getFormatterWithoutSymbol().format(item.getDiscountPerUnit())
+
+        htmlContent.getElementById(elementId).append(ItemPrintout.discountItemInHTML(itemNumber, description, quantity, unit, unitDiscount, discountQuantity))
+        consumedPageSpace += determineItemSpace("Discount", description, quantity, unit, unitDiscount, discountQuantity)
     }
 
     private static String generateElementID(int tableCount, ProductGroup productGroups){
         return productGroups.toString() + "-" + "product-items" + "-" + ++tableCount}
 
-    private void resetPageItemsCount(){ pageItemsCount = 1}
+    private void resetPageItemsCount(){ consumedPageSpace = 1}
 
     /**
      * Creates a new table element after a page break
      * @param elementId The id which will reference the new table
      */
     private void generateHTMLTableOnNextPage(String elementId){
         htmlContent.getElementById(elementId).append(ItemPrintout.pageBreak())
-        elementId = "product-items" + "-" + ++tableCount
-        htmlContent.getElementById("item-table-grid").append(ItemPrintout.createNewTable(elementId))
     }
 
-    private boolean isOverflowingPage() {
-        return pageItemsCount >= maxPageItems
+    private boolean isOverflowingPage(double elementSpace) {
```

Comment:
maybe rename to isItemOverflowingPage() or so. Because what you do is testing if the parameter given is causing an overflow.

Reply:
That makes sense good point 👍